handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -34,6 +35,9 @@
  * Triggers an {@link IdleStateEvent} when a {@link Channel} has not performed
  * read, write, or both operation for a while.
  *
+ * The {@link IdleStateHandler} should be placed as first {@link ChannelHandler}
```

Comment:
I'd argue that it depends on the use-case and is possibly dangerous advise. E.g. putting it before a decoder (server side) will expose the user to certain DoS attacks.